samza-api/src/main/java/org/apache/samza/operators/triggers/Triggers.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.operators.triggers;
+
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.operators.data.MessageEnvelope;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * API for specifying {@link Trigger}s for a {@link org.apache.samza.operators.windows.Window}.
+ *
+ * <p> The below example windows an input into tumbling windows of 10s, and emits early results periodically every 4s in
+ * processing time, or for every 50 messages. It also specifies that window results are accumulating.
+ *
+ * <pre> {@code
+ * MessageStream<> windowedStream = stream.window(
+ *   Windows.tumblingWindow(Duration.of(10, TimeUnit.SECONDS))
+ *     .setEarlyTrigger(Triggers.repeat(Triggers.any(Triggers.count(50), Triggers.timeSinceFirstMessage(Duration.of(4, TimeUnit.SECONDS))))))
+ *     .accumulateFiredPanes());
+ *   }
+ *  </pre>
```

Comment:
It looks like the closing pre is indented 2 spaces from the star while the opening pre is 1 space. Similarly, the closing brace for {@code should be at 1 indent from "*" too. This is probably already fixed in your reformat to indent code by 2.